overlord/snapstate/handlers.go

```diff
@@ -745,7 +807,11 @@ func (m *SnapManager) doUnlinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	// mark as inactive
 	snapst.Active = false
 	Set(st, snapsup.Name(), snapst)
-	return nil
+
+	if err := m.removeSnapContext(st, snapsup.Name()); err != nil {
```

Comment:
Besides the above concern, this one should also be moved to doDiscardSnap which is the place where we actually kill the left overs of the snap similar to that one.